src/scaffold/entry_type/integrity.rs

```diff
@@ -846,54 +855,62 @@ fn add_entry_type_to_validation_arms(
-                                                                    .eq(&String::from("Entry"))
-                                                                {
-                                                                    // Add new entry type to match arm
-                                                                    if find_ending_match_expr(
-                                                                        &mut op_entry_arm.body,
-                                                                    )
-                                                                    .is_none()
-                                                                    {
-                                                                        // Change empty invalid to match on entry_type
-                                                                        *op_entry_arm.body = syn::parse_str::<syn::Expr>("match original_app_entry {}")?;
-                                                                    }
-                                                                    // Add new entry type to match arm
-                                                                    if let Some(entry_type_match) =
-                                                                        find_ending_match_expr(
-                                                                            &mut op_entry_arm.body,
-                                                                        )
-                                                                    {
-                                                                        let new_arm: syn::Arm = syn::parse_str(
-                                                                            format!("EntryTypes::{}({}) => validate_delete_{}(action, original_action, {}),", 
-                                                                                entry_def.name.to_case(Case::Pascal),
-                                                                                entry_def.name.to_case(Case::Snake),
-                                                                                entry_def.name.to_case(Case::Snake),
-                                                                                entry_def.name.to_case(Case::Snake)
-                                                                            ).as_str()
-                                                                        )?;
-                                                                        entry_type_match
-                                                                            .arms
-                                                                            .push(new_arm);
-                                                                    }
+                                                if arm.body.as_ref().to_token_stream().to_string().contains("There are no entry types in this integrity zome") {
```

Comment:
Is there maybe another way to know that there are no entry types than relying on a string snippet here (if not, ignore this ocmment)? It just seems a bit prone to accidental failure. It's not obvious that the specific wording of the string is used here so it might easily happen that someone changes that string and things silently fail here.